paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc

```diff
@@ -0,0 +1,263 @@
+/* Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "mkldnn.hpp"
+#include "paddle/fluid/framework/op_registry.h"
+#include "paddle/fluid/framework/tensor.h"
+#include "paddle/fluid/operators/math/blas.h"
+#include "paddle/fluid/platform/mkldnn_helper.h"
+
+namespace paddle {
+namespace operators {
+
+using dnnl::memory;
+using dnnl::primitive;
+using platform::to_void_cast;
+using framework::DataLayout;
+using platform::GetMKLDNNFormat;
+using platform::MKLDNNGetDataType;
+using platform::MKLDNNDeviceContext;
+using framework::ExecutionContext;
+using Tensor = framework::Tensor;
+
+// Get row matrix shape from a vector shape. If the rank of x_dim > 1, the
+// original x_dim is returned.
+static framework::DDim RowMatrixFromVector(const framework::DDim& x_dim) {
```

Comment:
The name is a bit misleading, as the function is about the shape, not about the matrix.
Same below

Reply:
What other name do you suggest instead? I think it conveys the meaning that it accepts a vector of size N and converts it into a matrix of size 1 x N. Similarly for the other function.

Reply:
Maybe `RowMatrixDimsFromVector`, or `VectorDims2RowMatrixDims`?
It is only a suggestion.

Reply:
But it's the same function as already exists here:
https://github.com/PaddlePaddle/Paddle/pull/23395/files/471e894b65b0cf2db48de2ee2b0c330d3cbd7815#diff-126803a76a1b0cbf6198f30e23058f8dL39